models/groups.go

```diff
@@ -194,20 +194,20 @@ func (g *CreateUpdateGroup) ValidateCreateUpdateGroupRequest(ctx context.Context
 
 func (c *CreateUpdateGroup) BuildCreateGroupInput(userPoolId *string) *cognitoidentityprovider.CreateGroupInput {
 	return &cognitoidentityprovider.CreateGroupInput{
-		Description: c.Description,
-		GroupName:   c.GroupName,
+		Description: c.Name,
+		GroupName:   c.ID,
 		Precedence:  c.Precedence,
 		UserPoolId:  userPoolId,
 	}
 }
 
 // BuildUpdateGroupInput builds a correctly populated UpdateGroupInput object using Groups values
 func (g *CreateUpdateGroup) BuildUpdateGroupInput(userPoolId string) *cognitoidentityprovider.UpdateGroupInput {
-	return &cognitoidentityprovider.UpdateGroupInput {
-		GroupName: g.GroupName,
-		Description: g.Description,
-		Precedence: g.Precedence,
-		UserPoolId: &userPoolId,
+	return &cognitoidentityprovider.UpdateGroupInput{
+		GroupName:   g.Name,
```

Comment:
I believe this line should likely be `g.ID`:

```suggestion
		GroupName:   g.ID,
```